Tests/Services/TasksServiceTests.cs

```diff
@@ -415,4 +408,412 @@ await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+                         Tasks =
+                         {
+                           new TaskSummary
+                           {
+                             Id        = "taskId1",
+                             Status    = (ArmoniK.Api.gRPC.V1.TaskStatus)TaskStatus.Completed,
+                             CreatedAt = Timestamp.FromDateTime(DateTime.UtcNow),
+                             StartedAt = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-5)),
+                             EndedAt   = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-1)),
+                           },
+                           new TaskSummary
+                           {
+                             Id        = "taskId2",
+                             Status    = (ArmoniK.Api.gRPC.V1.TaskStatus)TaskStatus.Cancelling,
+                             CreatedAt = Timestamp.FromDateTime(DateTime.UtcNow),
+                             StartedAt = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-10)),
+                             EndedAt   = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-5)),
+                           },
+                         },
+                         Total = 2,
+                       };
+
+    var mockInvoker = new Mock<CallInvoker>();
+    var callInvoker = mockInvoker.SetupAsyncUnaryCallInvokerMock<ListTasksRequest, ListTasksResponse>(taskResponse);
+
+    var taskService = MockHelper.GetTasksServiceMock(callInvoker);
+
+    var paginationOptions = new TaskPagination
+                            {
+                              Page          = 1,
+                              PageSize      = 10,
+                              SortDirection = SortDirection.Asc,
+                              Total         = 2,
+                              Filter        = new Filters(),
+                            };
+
+    var result = await taskService.ListTasksAsync(paginationOptions)
+                                  .ToListAsync();
+
+    ClassicAssert.IsNotNull(result);
+    ClassicAssert.AreEqual(1,
+                           result.Count);
+    ClassicAssert.AreEqual(2,
+                           result[0].TotalTasks);
+
+    var tasksData = result[0]
+                    .TasksData.ToList();
+    ClassicAssert.AreEqual(2,
+                           tasksData.Count);
+
+    var expectedTaskIds = new List<string>
+                          {
+                            "taskId1",
+                            "taskId2",
+                          };
+    var expectedStatuses = new List<TaskStatus>
+                           {
+                             TaskStatus.Completed,
+                             TaskStatus.Cancelling,
+                           };
+
+    for (var i = 0; i < tasksData.Count; i++)
+    {
+      ClassicAssert.AreEqual(expectedTaskIds[i],
+                             tasksData[i].Item1);
+      ClassicAssert.AreEqual(expectedStatuses[i],
+                             tasksData[i].Item2);
+    }
```

Comment:
Please use the [Contraint model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html) of assertions.